validator/client/src/main/java/tech/pegasys/teku/validator/client/RuntimeProposerConfig.java

```diff
@@ -68,20 +72,45 @@ public Optional<Eth1Address> getEth1AddressForPubKey(final BLSPublicKey pubKey)
     if (config == null) {
       return Optional.empty();
     }
-    return Optional.of(config.getFeeRecipient());
+    return config.getFeeRecipient();
   }
 
-  public synchronized void addOrUpdate(
+  public Optional<UInt64> getGasLimitForPubKey(final BLSPublicKey publicKey) {
+    final Config config = proposerConfigMap.get(publicKey);
+    if (config == null) {
+      return Optional.empty();
+    }
+    return config.getGasLimit();
```

Comment:
may look neater if we 
```
final Optional<Config> maybeConfig = Optional.ofNullable(proposerConfigMap.get(publicKey));
return maybeConfig.map(Config::getGasLimit);
```

Reply:
You mean `return maybeConfig.flatMap(Config::getGasLimit);` I guess :)